common/src/test/kotlin/streams/service/sink/strategy/CUDIngestionStrategyTest.kt

```diff
@@ -0,0 +1,572 @@
+                |DETACH DELETE n
+            """.trimMargin(), nodeDeleteEvents)
+        assertEquals(2, deleteNodeFooBarDetach.events.size)
+        assertNodeEventsContainsKey(deleteNodeFooBarDetach, "key")
+        val deleteNodeFooBarLabel = findEventByQuery("""
+                |${StreamsUtils.UNWIND}
+                |MATCH (n:Foo:Bar:Label {key: event.${CUDIngestionStrategy.ID_KEY}.key})
+                |DELETE n
+            """.trimMargin(), nodeDeleteEvents)
+        assertEquals(3, deleteNodeFooBarLabel.events.size)
+        assertNodeEventsContainsKey(deleteNodeFooBarLabel, "key")
+        val deleteNodeFooBarLabelDetach = findEventByQuery("""
+                |${StreamsUtils.UNWIND}
+                |MATCH (n:Foo:Bar:Label {key: event.${CUDIngestionStrategy.ID_KEY}.key})
+                |DETACH DELETE n
+            """.trimMargin(), nodeDeleteEvents)
+        assertEquals(2, deleteNodeFooBarLabelDetach.events.size)
+        assertNodeEventsContainsKey(deleteNodeFooBarLabelDetach, "key")
+    }
+
+    @Test
+    fun `should delete nodes with internal id reference`() {
+        // given
+        val detachMarkers = listOf(1, 3, 8, 10)
+        val list = (1..10).map {
+            val labels = if (it % 2 == 0) listOf("Foo", "Bar") else listOf("Foo", "Bar", "Label")
+            val detach = it in detachMarkers
+            val properties = mapOf("foo" to "foo-value-$it", "id" to it)
+            val cudNode = CUDNode(op = CUDOperations.delete,
+                    labels = labels,
+                    ids = mapOf("_id" to it),
+                    properties = properties,
+                    detach = detach)
+            StreamsSinkEntity(null, cudNode)
+        }
+
+        // when
+        val cudQueryStrategy = CUDIngestionStrategy()
+        val nodeEvents = cudQueryStrategy.mergeNodeEvents(list)
+        val nodeDeleteEvents = cudQueryStrategy.deleteNodeEvents(list)
+
+        val relationshipEvents = cudQueryStrategy.mergeRelationshipEvents(list)
+        val relationshipDeleteEvents = cudQueryStrategy.deleteRelationshipEvents(list)
+
+        // then
+        assertEquals(emptyList(), nodeEvents)
+        assertEquals(emptyList(), relationshipEvents)
+        assertEquals(emptyList(), relationshipDeleteEvents)
+
+        assertEquals(4, nodeDeleteEvents.size)
+        assertEquals(10, nodeDeleteEvents.map { it.events.size }.sum())
+        val deleteNodeFooBar = findEventByQuery("""
+                |${StreamsUtils.UNWIND}
+                |MATCH (n:Foo:Bar) WHERE id(n) = event.${CUDIngestionStrategy.ID_KEY}._id
+                |DELETE n
+            """.trimMargin(), nodeDeleteEvents)
+        assertEquals(3, deleteNodeFooBar.events.size)
+        val key = "_id"
+        assertNodeEventsContainsKey(deleteNodeFooBar, key)
+        val deleteNodeFooBarDetach = findEventByQuery("""
+                |${StreamsUtils.UNWIND}
+                |MATCH (n:Foo:Bar) WHERE id(n) = event.${CUDIngestionStrategy.ID_KEY}._id
```

Comment:
See earlier comment.  If we generate cypher like this and _id = "banana", then the plugin will successfully process the message, nothing will happen, and the user will think the plugin is very mysterious.  :)

Some upfront message validation may be necessary